mllib/src/main/scala/org/apache/spark/ml/tree/impl/RandomForest.scala

```diff
@@ -1025,7 +1048,9 @@ private[ml] object RandomForest extends Logging {
       new mutable.HashMap[Int, mutable.HashMap[Int, NodeIndexInfo]]()
     var memUsage: Long = 0L
     var numNodesInGroup = 0
-    while (nodeQueue.nonEmpty && memUsage < maxMemoryUsage) {
+    // If maxMemoryInMB is set very small, we want to still try to split 1 node,
+    // so we allow one iteration if memUsage == 0.
+    while (nodeQueue.nonEmpty && (memUsage < maxMemoryUsage || memUsage == 0)) {
```

Comment:
The idea is to train at minimum one node, regardless of memory constraints, I believe. Checking if `memUsage == 0` works, but a more direct way of saying this might be `mutableNodesForGroup.isEmpty`. The only reason I point this out is because, as I read the code, I was wondering if you could ever add a node without increasing `memUsage` (can `nodeMemUsage` ever be zero -- it cannot). So, it might make the code more understandable/explicit, but I don't feel too strongly about it.